File: backend/src/main/java/com/grdp/studio/softwareintegration/support/SystemAnalysisParameters.java
package com.grdp.studio.softwareintegration.support;

import tools.jackson.databind.JsonNode;

public final class SystemAnalysisParameters {
    private SystemAnalysisParameters() {}

    public static boolean valid(JsonNode value) {
        if (value == null || !value.isObject() || value.size() != 6
                || !"pipesim-system-analysis-parameters/1".equals(value.path("schemaVersion").asText())
                || !"Well".equals(value.path("producer").asText())
                || !"liquidFlowRate".equals(value.path("scanVariable").asText())
                || !value.path("branchTerminator").isTextual()
                || !safeName(value.path("branchTerminator").asText())
                || !number(value, "outletPressurePsi", 0, 100000)
                || !value.path("values").isArray()
                || value.path("values").size() < 2 || value.path("values").size() > 8) return false;
        double previous = Double.NEGATIVE_INFINITY;
        for (JsonNode item : value.path("values")) {
            if (!item.isNumber() || !Double.isFinite(item.asDouble()) || item.asDouble() <= 0
                    || item.asDouble() <= previous || item.asDouble() > 1000000000) return false;
            previous = item.asDouble();
        }
        return true;
    }

    private static boolean number(JsonNode object, String field, double lowerExclusive, double upperInclusive) {
        JsonNode value = object.get(field);
        return value != null && value.isNumber() && Double.isFinite(value.asDouble())
                && value.asDouble() > lowerExclusive && value.asDouble() <= upperInclusive;
    }

    private static boolean safeName(String value) {
        return value != null && !value.isBlank() && value.length() <= 255
                && value.indexOf('\u0000') < 0 && value.indexOf('/') < 0
                && value.indexOf('\\') < 0 && !value.contains("..");
    }
}

File: worker/Execution/SystemAnalysisParameters.cs
using System.Text.Json;

namespace Grdp.SoftwareIntegration.Worker.Execution;

internal static class SystemAnalysisParameters
{
    internal static bool Valid(JsonElement value)
    {
        if (value.ValueKind != JsonValueKind.Object
            || value.EnumerateObject().Count() != 6
            || !Text(value, "schemaVersion", "pipesim-system-analysis-parameters/1")
            || !Text(value, "producer", "Well")
            || !Text(value, "scanVariable", "liquidFlowRate")
            || !value.TryGetProperty("branchTerminator", out var branch)
            || !SafeName(branch)
            || !Number(value, "outletPressurePsi", 0, 100000)
            || !value.TryGetProperty("values", out var values)
            || values.ValueKind != JsonValueKind.Array
            || values.GetArrayLength() is < 2 or > 8)
        {
            return false;
        }

        var previous = double.NegativeInfinity;
        foreach (var item in values.EnumerateArray())
        {
            if (item.ValueKind != JsonValueKind.Number || !item.TryGetDouble(out var number)
                || !double.IsFinite(number) || number <= 0 || number <= previous || number > 1000000000)
            {
                return false;
            }
            previous = number;
        }
        return true;
    }

    private static bool Text(JsonElement value, string name, string expected) =>
        value.TryGetProperty(name, out var item) && item.ValueKind == JsonValueKind.String && item.GetString() == expected;

    private static bool SafeName(JsonElement value)
    {
        if (value.ValueKind != JsonValueKind.String) return false;
        var text = value.GetString();
        return !string.IsNullOrWhiteSpace(text) && text.Length <= 255
            && !text.Contains("..", StringComparison.Ordinal)
            && text.IndexOfAny(['/', '\\', '\0']) < 0;
    }

    private static bool Number(JsonElement value, string name, double lowerExclusive, double upperInclusive) =>
        value.TryGetProperty(name, out var item) && item.ValueKind == JsonValueKind.Number
        && item.TryGetDouble(out var number) && double.IsFinite(number)
        && number > lowerExclusive && number <= upperInclusive;
}
